src/game/objects/player.ts

```diff
@@ -124,11 +131,15 @@ export abstract class Character {
         if (this.itemDuration.boost > 0) {
             this.itemDuration.boost -= 1
             if (this.itemDuration.boost <= 0) {
-                this.speed -= ITEM.SPEED_UP // 부스터 효과 종료
                 this.itemDuration.boost = 0
             }
         }
 
+        if (this.itemDuration.boost > 0) {
+            calculatedSpeed += ITEM.SPEED_UP
+        }
+        this.speed = Math.max(this.basespeed / 2, calculatedSpeed)
```

Comment:
Q1. 위쪽 `//부스터 지속 시간 처리` 로직이 간소화될 수 있을거 같아요

 ```suggestion
        if (this.itemDuration.boost > 0) {
            this.itemDuration.boost -= 1
        }
```

Q2. 같이 얘기한거 : 부스트 적용 시, 최대속도 경험할 수 있도록